### ISSUE TITLE
error after update to 1.3.6

### ISSUE THREAD

**Issue by marek-l** [jackalope/jackalope#352]
Hi,

after update to version 1.3.6 we got error: Attempting to get the path for a non existent selector: nt:file
With version 1.3.5 is everything ok, any ideas?

thanks

**Comment by dbu** [comment]
could it be this fix: https://github.com/jackalope/jackalope/commit/cae9adc8ab775f3f5276dcd6249e5b7ecb375925

can you try if you revert that line if that fixes the problem for you? if it does, we need to figure out in which situtations what case would be correct.

**Comment by marek-l** [comment]
> could it be this fix: [cae9adc](https://github.com/jackalope/jackalope/commit/cae9adc8ab775f3f5276dcd6249e5b7ecb375925) ?
> 
> can you try if you revert that line if that fixes the problem for you? if it does, we need to figure out in which situtations what case would be correct.

yes, after changing code back to:
`$this->defaultSelectorName = $selectorName;`
everything works fine

**Comment by dbu** [comment]
@marek-l could you try the code snippet from #351 in your installation with just the reverted line to know if that problem would re-appear or not?

```php
$result = $qm->createQuery(
  "SELECT [jcr:primaryType] FROM [rep:root]",
  "JCR-SQL2"
)
->execute();

foreach ($result as $row) {
  echo $row->getValue("jcr:primaryType") . "\n";
}
```

@marek-l on which jackalope flavour are you, doctrine dbal or jackrabbit?
@middlebrain which jackalope version are you using, doctrine dbal or jackrabbit? do you have an idea whats happening here?

@marek-l best lock to version 1.3.5 until we figured out what is happening here.

**Comment by middlebrain** [comment]
> @middlebrain which jackalope version are you using, doctrine dbal or jackrabbit?

jackrabbit

> do you have an idea whats happening here?

Seems, that my fix when using aliases (```SELECT * FROM [ns:foo] AS f INNER JOIN [ns:bar] AS b ON ...```) sometimes sets no or a wrong ```defaultSelectorName``` (for example ```ns:foo``` instead of ```f```). But at the moment, I don't know how to fix this. I guess the best thing to do is revert my 'fix'.

**Comment by marek-l** [comment]
> @marek-l could you try the code snippet from #351 in your installation with just the reverted line to know if that problem would re-appear or not?
> 
> ```
> $result = $qm->createQuery(
>   "SELECT [jcr:primaryType] FROM [rep:root]",
>   "JCR-SQL2"
> )
> ->execute();
> 
> foreach ($result as $row) {
>   echo $row->getValue("jcr:primaryType") . "\n";
> }
> ```
> 
> @marek-l on which jackalope flavour are you, doctrine dbal or jackrabbit?
> @middlebrain which jackalope version are you using, doctrine dbal or jackrabbit? do you have an idea whats happening here?
> 
> @marek-l best lock to version 1.3.5 until we figured out what is happening here.

$row is instance of Doctrine\\ODM\\PHPCR\\Document\\Generic and this class does not have method getValue()

we are on jackrabbit

**Comment by dbu** [comment]
fixed in #353
